src/dbhelper.py

```diff
@@ -0,0 +1,107 @@
+from PySide import QtSql
+from PySide.QtSql import QSqlQuery
+from passwords import DB_SERVER, DB_PORT, DB_LOGIN, DB_PASSWORD, DB_TABLE
+import logging
+
+# Object to wrap a database connection and provide some vaguely handy utilities.
+_logger = logging.getLogger(__name__)
+
+# Open the database connection. when the module is first touched...
+_db = QtSql.QSqlDatabase("QMYSQL")
+_db.setHostName(DB_SERVER)
+_db.setPort(DB_PORT)
+
+_db.setDatabaseName(DB_TABLE)
+_db.setUserName(DB_LOGIN)
+_db.setPassword(DB_PASSWORD)
+
+_db.setConnectOptions("MYSQL_OPT_RECONNECT=1")
+
+if not _db.open():
+    _logger.error(_db.lastError().text())
+    raise RuntimeError
+
+# A cache of prepared queries, keyed by SQL string. Allows reuse of prepared queries.
+_prepared_queries = {}
```

Comment:
Where exactly are we opening and closing the database? From my findings, we keep one persistent connection. At least I cannot find any calls to `db.close()`.

If the IPC call actually blocks the entire eventloop, this would be bad -- but I'm not convinced that that is the case. Given that it does not, it is a cost we can live with in terms of latency, and it can potentially improve overall responsiveness of the server by allowing other events to be processed in the meantime.